helpers/block_visitor.go

```diff
@@ -0,0 +1,43 @@
+package helpers
+
+import (
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-fetcher"
+	"github.com/ipld/go-ipld-prime"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+)
+
+// BlockResult specifies a node at the top of a block boundary
+type BlockResult struct {
```

Comment:
Is this really necessary? It seem like you might want the path. And it seems a shame to have to wrap an existing `FetchCallback` when you want to use it in an OnBlocks context. My vote would be to drop this and have OnBlocks and OnUniqueBlocks take a `FetchCallback`.

Reply:
the reason is at least for OnUniqueBlocks, two paths may appear for the block and you'd miss the second one.